domain/src/main/java/org/oppia/android/domain/exploration/ExplorationProgressListenerImpl.kt

```diff
@@ -0,0 +1,27 @@
+package org.oppia.android.domain.exploration
+
+import org.oppia.android.app.model.ProfileId
+import org.oppia.android.domain.oppialogger.analytics.ApplicationLifecycleListener
+import javax.inject.Inject
+import javax.inject.Singleton
+
+/** Implementation of [ExplorationProgressListener] */
```

Comment:
Calling listener implementations generic "impl"s poses a problem of vagueness. Listeners are meant to exist more than once, so each implementation ought to be specific to whichever domain purpose it's fulfilling (versus utilities which generally only have one implementation). I suggest renaming this class to be more specific as to what it's responsible for rather than just naming it an implementation of the listener class (especially since it's an impl of multiple listeners, so there's clearly more specificity possible here).

Reply:
Since the purpose here is to start and stop the session timer, I have renamed it to `ExplorationSessionTimerController`